pkg/adaptation/plugin.go

```diff
@@ -418,6 +418,7 @@ func (p *plugin) synchronize(ctx context.Context, pods []*PodSandbox, containers
 	}
 	rpl, err := p.stub.Synchronize(ctx, req)
 	if err != nil {
+		p.close()
```

Comment:
nit.. rename this func to initialSynchronizeOnConnect
same on above log.Info.. "synchronizing plugin on connect %s"

Reply:
or will it also be used in catchup scenarios ..